client/src/rpc_client.rs

```diff
@@ -1012,49 +1012,55 @@ impl RpcClient {
         &self,
         transaction: &Transaction,
     ) -> ClientResult<Signature> {
```

Comment:
I'm a little uncomfortable about changing `send_and_confirm_transaction()` to infinitely retry sending the transaction.  At the very least I'd like to discuss that change in a PR that claims to just be performing a refactoring chore.  

If you'd like to proceed with the transaction retrying, I suggest that:
1. This PR be reduced to refactoring while preserving the existing send_and_confirm_transaction logic
2. A new PR be created to add retries

Thanks for the contribution though!

Reply:
Oh, yeah, default value of `SEND_RETRIES` must be `1`, not `MAX`.

https://github.com/solana-labs/solana/pull/18389 this PR contains simple refactoring.